eventstream.go

```diff
@@ -356,4 +362,267 @@ type Client interface {
 	Publish(publishBuilder *PublishBuilder) error
 	PublishSync(publishBuilder *PublishBuilder) error
 	Register(subscribeBuilder *SubscribeBuilder) error
+	PublishAuditLog(auditLogBuilder *AuditLogBuilder) error
+}
+
+type AuditLog struct {
+	ID              string           `json:"id,omitempty"`
+	Category        string           `json:"category,omitempty"`
+	ActionName      string           `json:"actionName,omitempty"`
+	Timestamp       int64            `json:"timestamp,omitempty"`
+	IP              string           `json:"ip,omitempty"`
+	Actor           string           `json:"actor,omitempty"`
+	ActorType       string           `json:"actorType,omitempty"`
+	ClientID        string           `json:"clientId,omitempty"`
+	ActorNamespace  string           `json:"actorNamespace,omitempty"`
+	ObjectIDs       []string         `json:"objectIds,omitempty"`
+	ObjectNamespace string           `json:"objectNamespace"`
+	TargetUserIDs   []string         `json:"targetUserIds,omitempty"`
+	DeviceID        string           `json:"deviceId,omitempty"`
+	Payload         *AuditLogPayload `json:"payload,omitempty"`
+}
+
+type EventBuilder interface {
+	Build() (kafka.Message, interface{}, error)
+	Validate() error
+	GetTopics() []string
+	GetEventName() string
+	GetContext() context.Context
+	GetErrorCallback() errorCallbackFunc
+}
+
+type errorCallbackFunc func(event interface{}, err error)
+
+type AuditLogBuilder struct {
+	topics          []string
```

Comment:
does that mean we need to publish them on multiple topics? who will consume them?